### PR TITLE
Create New Content Modal

### PR DESCRIPTION
**Pull request by mateoclarke**
Closes https://github.com/cityofaustin/techstack/issues/415

Still need to style this and refactor some code but the functionality should be here.

 
aug-08-2018 22-25-20

### CODE REVIEW

File: joplin/base/templatetags/joplin_tags.py [hunk 1]
```diff
@@ -28,5 +35,26 @@ def get_preview_url(*args, **kwargs):
 @register.simple_tag
 def get_style_guide_url(*args, **kwargs):
   content_type = kwargs['content_type'].name
-
   return os.environ['STYLEGUIDE_URL'] + STYLEGUIDE_PAGES[content_type]
+
+@register.inclusion_tag('wagtailadmin/themes.html', takes_context=True)
```
Comment (briaguya) [inline joplin/base/templatetags/joplin_tags.py]: hooray for tags!

File: joplin/base/wagtail_hooks.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 def global_css():
     urls = [
         static('css/admin.css'),
+        static('css/create_content_modal.css'),
```
Comment (briaguya) [inline joplin/base/wagtail_hooks.py]: more motivation for me to get my webpack PR out

File: joplin/templates/wagtailadmin/base.html [hunk 3]
```diff
@@ -9,7 +9,10 @@
 
     <div class="header">
         <div>
-            <img class="wagtail-logo wagtail-logo__full" src="{% static 'images/logo.svg' %}" alt="Welcome to Pearl" width="50" />
+            <img class="wagtail-logo wagtail-logo__full"
+              src="{% static 'images/coa_seal_transparent_white.png' %}"
+              alt="Welcome to Wagtail CMS" width="60"
```
Comment (briaguya) [inline joplin/templates/wagtailadmin/base.html]: do we want it to say Wagtail? also I kinda miss the chicken but I guess we can be official

File: joplin/templates/wagtailadmin/pages/create.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        Additional js from widgets media. Allows for custom widgets in admin panel.
+    {% endcomment %}
+    {{ edit_handler.form.media.js }}
+
+    {% comment %}
+        Additional HTML code that edit handlers define through 'html_declarations'. (Technically this isn't Javascript, but it will generally be data that exists for Javascript to work with...)
+    {% endcomment %}
+    {{ edit_handler.html_declarations }}
+
+    <script>
```
Comment (briaguya) [inline joplin/templates/wagtailadmin/pages/create.html]: ooh can't wait to webpack it all up!!!

File: joplin/templates/wagtailadmin/pages/create.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        $(function(){
+            $('#page-edit-form .tab-content section.active input').first().trigger('focus');
+            /* Make user confirm before leaving the editor if there are unsaved changes */
+            {% trans "This page has unsaved changes." as confirmation_message %}
+            enableDirtyFormCheck(
```
Comment (briaguya) [inline joplin/templates/wagtailadmin/pages/create.html]: this wasn't already there?

File: joplin/templates/wagtailadmin/pages/listing/_create_content_modal.html [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+<div class="modal fade" id="createNewContentModal" tabindex="-1" role="dialog" aria-labelledby="createNewContentModalLabel">
+  <div class="js-create-content-wizard-form">
+    <div class="modal-dialog" role="document">
+      <div class="modal-content">
+        <div class="modal-header">
+          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <div class="modal-body">
```
Comment (briaguya) [inline joplin/templates/wagtailadmin/pages/listing/_create_content_modal.html]: these make me miss react

File: joplin/templates/wagtailadmin/pages/listing/_create_content_modal.html [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+          </div>
+
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script>
+  $(document).ready(function() {
```
Comment (briaguya) [inline joplin/templates/wagtailadmin/pages/listing/_create_content_modal.html]: can't wait for the webpack!

File: joplin/templates/wagtailadmin/themes.html [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<h1>Themes with nested topics 💩</h1>
```
Comment (briaguya) [inline joplin/templates/wagtailadmin/themes.html]: Nice! 💩

File: joplin/templates/wagtailadmin/base.html (hunk 3 above)
Comment (mateoclarke) [reply joplin/templates/wagtailadmin/base.html]: I'm not sure. I was between that and Joplin CMS. But if there is a content editor or contract dev in the future, they could at least Google: "Wagtail CMS" and find their way to some docs and tutorials.

I'm kinda against the use of the love chicken outside our team. It's internal branding for our office which make us hypocritical whenever we tell other departments their brand doesn't matter. If this is a CMS for the whole city, we should use the City branding.

File: joplin/templates/wagtailadmin/pages/create.html (hunk 5 above)
Comment (mateoclarke) [reply joplin/templates/wagtailadmin/pages/create.html]: no, this file extends the `joplin/templates/wagtailadmin/pages/create.html` file from Wagtail core

File: joplin/templates/wagtailadmin/base.html (hunk 3 above)
Comment (briaguya) [reply joplin/templates/wagtailadmin/base.html]: I completely agree on the branding. Doesn't make me not miss the chicken though.

File: joplin/templates/wagtailadmin/pages/create.html (hunk 5 above)
Comment (briaguya) [reply joplin/templates/wagtailadmin/pages/create.html]: hmm, extending generally means you still have the same page there but are replacing some blocks, where as if you don't use the extend tag then you can replace the entire page. 

I'd think the block.super should be able to get enableDirtyFormCheck in there.

File: joplin/templates/wagtailadmin/base.html (hunk 3 above)
Comment (toribr) [reply joplin/templates/wagtailadmin/base.html]: I agree too, but the City Seal is too small to be legible, so also isn't a great choice. Maybe we can get permission to use a variation of this, which is far more web friendly?
image

Comment (toribr) [reply joplin/templates/wagtailadmin/base.html]: I never liked the love chicken, so I'm not arguing with you on that point. However, I disagree that it doesn't matter if it's not legible. We're losing an opportunity to show them clean, well-designed internal tools. Internal audiences deserve good design too. We just don't have anything better at the moment.

File: joplin/base/templatetags/joplin_tags.py [hunk 9]
```diff
@@ -28,5 +35,26 @@ def get_preview_url(*args, **kwargs):
...
+def themes(context):
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply joplin/base/templatetags/joplin_tags.py]: Yeah, good call. That name is too broad

File: joplin/base/templatetags/joplin_tags.py [hunk 10]
```diff
@@ -1,6 +1,13 @@
 from django import template
 import graphene
 import os
+from base.models import Topic, Theme
+# from base.models import Theme
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply joplin/base/templatetags/joplin_tags.py]: done in the next branch

File: joplin/templates/wagtailadmin/pages/create.html (hunk 5 above)
Comment (mateoclarke) [reply joplin/templates/wagtailadmin/pages/create.html]: thanks for explaining. fixed in the next branch

File: joplin/templates/wagtailadmin/pages/listing/_create_content_modal.html [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+            </div>
+          </div>
+
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script>
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply joplin/templates/wagtailadmin/pages/listing/_create_content_modal.html]: I thinks Brian's webpacking work is going to solve this differently but I went ahead and made this change in the branch i'm working on

File: joplin/templates/wagtailadmin/pages/listing/_create_content_modal.html [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+  $(document).ready(function() {
+
+    var contentWizardState = {
+      type: '',
+      title: '',
+      topic: '',
+      activeStep: 0,
+    }
+
+    var contentWizardModule = {
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply joplin/templates/wagtailadmin/pages/listing/_create_content_modal.html]: Let's set aside time this week to pair on refactoring here like you suggest.

File: joplin/templates/wagtailadmin/pages/listing/_list.html [hunk 13]
```diff
@@ -72,3 +75,5 @@ <h1>Content</h1>
         {% endif %}
     </tbody>
 </table>
+
+{% include "./_create_content_modal.html" %}
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply joplin/templates/wagtailadmin/pages/listing/_list.html]: no it works without the prefix. I was just following the pattern of naming for the two files adjacent to it. 

I'm down for creating separate templates. Can we table that as tech debt for once we get past the MVP? Can you create an issue?

**Pull request merged**